### ISSUE TITLE
OS X Command+V

### ISSUE THREAD

**Issue by neseleznev** [mapbox/mapbox-gl-geocoder#140]
_geocode is not invoking while user pastes searchInput via Cmd+V on OS X.

The issue is supposed to by fixed in the following line
https://github.com/mapbox/mapbox-gl-geocoder/blob/53d06d871574a227a6b414f3370247ea16e29238/lib/index.js#L119
since Cmd is a metaKey.

Probably, it could be checked by something like e.code === 'MetaLeft' || e.code === 'MetaRight' ||  e.code === 'OSLeft' || e.code === 'OSRight';

I'm not a web developer, correct me if I'm wrong

**Comment by andrewharvey** [comment]
Which version of mapbox-gl-geocoder are you using?

Does it not open the suggestions list after pasting or just it won't paste into the input at all?

The latest version fixed paste events with https://github.com/tristen/suggestions/pull/17 see https://github.com/mapbox/mapbox-gl-geocoder/blob/master/CHANGELOG.md could you confirm you're still seeing this with v2.1.2?

**Comment by ingalls** [comment]
Stale - Feel free to reopen if you are still having problems.

**Comment by matrad** [comment]
Still seeing this issue on v2.2.0 with Chrome 67.0.3396.99 x64 (newest build) and OSX 10.13.3

**Comment by gtk-hs** [comment]
Hey all! I'm currently seeing this issue on modern FF(68.0 )/chrome(75.0.3770.142)/Safari(12.1.2) OS X on our production site, and even on the mapbox website: 

https://docs.mapbox.com/help/tutorials/local-search-geocoding-api/

I've tried manually dispatching the change/keyup/keydown event on the target input field, but can't get it to populate the geocode dropdown UI.

This can be a painful UX experience for our users that are confused as to why pasting a valid address returns no response in the UI.

If you can get the dropdown to trigger on paste, that would be ideal, as users will see immediate feedback and pick the proper result. Also if hitting the enter key while focused on that dropdown kicked off the geocoder dropdown UI I think that would help as well (although hitting any OTHER key does appear to have the desired effect). While I'm at it I feel clicking the "search" svg icon when text is pasted in should probably kick off that UI as well (although it might be unnecessary if the paste event worked in the first place).

In the meanwhile, is there anyway I can programmatically kick off the dropdown of geocode results so I can reduce friction on end users who are having a hard time figuring this out?

**Comment by Grsmto** [comment]
@gtk-hs did you figure out a workaround for this? Or is there a fork available with a fix?

**Comment by gtk-hs** [comment]
@Grsmto Nope, paste still doesn't trigger the geocoder dropdown UI! We've just told support to tell users not to paste into that field (it hurts that this is true).

Please bug the developer, but this bug has been in the wild for two years so it might be time to start looking for a better supported geocoding solution.

**Comment by andrewharvey** [comment]
> Please bug the developer, but this bug has been in the wild for two years so it might be time to start looking for a better supported geocoding solution.

This is an open source project, please feel free to submit a PR if you manage to fix it.

**Comment by gtk-hs** [comment]
@andrewharvey my apologies, I must have confused this project with an enterprise product sold at www.mapbox.com, I'll contact their support team directly, sorry for any confusion!

**Comment by yapalexei** [comment]
For those still looking for options I had come up with one for our product that works for us. One could add a 'paste' || 'input' event listener to the input (search) field.

something like:
```
map.on("load", function() {

    const mapboxGeocoder = new MapboxGeocoder({
        accessToken: mapboxgl.accessToken,
        mapboxgl: mapboxgl
    });

    map.addControl(mapboxGeocoder);
    // The instance of MapboxGeocoder has a container property that you can query for the input field.
    // Although, document.querySelector('input.mapboxgl-ctrl-geocoder--input') could work too
    const inputField = mapboxGeocoder.container.querySelector('input.mapboxgl-ctrl-geocoder--input');

    if (!inputField) return

    inputField.addEventListener('paste', () => {
        const text = (event.clipboardData || window.clipboardData).getData('text');
        if (text && text.length > 2) {
            // This is the magic right here.
            mapboxGeocoder.query(text);
        }
    });
});
```

The only caveat is the call `mapboxGeocoder.query(text)` opens the dropdown and then takes you to the first location listed; though the drop down remains open.

**Comment by yapalexei** [comment]
Also here's a working example: https://codepen.io/yapalexei/pen/poozzaO?editors=0010

**Comment by andrewharvey** [comment]
@yapalexei This looks like the solution proposed in #300, if you wanted to see this fixed in core it would be great if you could test out #300 to see if it works for you.

**Comment by yapalexei** [comment]
@andrewharvey you're right! I should have checked that PR. I will check it out.

**Comment by scottsfarley93** [comment]
this was fixed in #300 and will be available in the upcoming v4.5.0 release.

**Comment by Pritzomania** [comment]
> For those still looking for options I had come up with one for our product that works for us. One could add a 'paste' || 'input' event listener to the input (search) field.
> 
> something like:
> 
> ```
> map.on("load", function() {
> 
>     const mapboxGeocoder = new MapboxGeocoder({
>         accessToken: mapboxgl.accessToken,
>         mapboxgl: mapboxgl
>     });
> 
>     map.addControl(mapboxGeocoder);
>     // The instance of MapboxGeocoder has a container property that you can query for the input field.
>     // Although, document.querySelector('input.mapboxgl-ctrl-geocoder--input') could work too
>     const inputField = mapboxGeocoder.container.querySelector('input.mapboxgl-ctrl-geocoder--input');
> 
>     if (!inputField) return
> 
>     inputField.addEventListener('paste', () => {
>         const text = (event.clipboardData || window.clipboardData).getData('text');
>         if (text && text.length > 2) {
>             // This is the magic right here.
>             mapboxGeocoder.query(text);
>         }
>     });
> });
> ```
> 
> The only caveat is the call `mapboxGeocoder.query(text)` opens the dropdown and then takes you to the first location listed; though the drop down remains open.

@yapalexei  "...though the drop down remains open" - Any idea on the best way to close the dropdown after triggering the _mapboxGeocoder.query_ method?
